docs/hoon/demo.md

```diff
@@ -42,22 +48,28 @@ Line 2 declares a variable, `count`, with initial value `1`.
 
 Line 3 begins a loop.
 
-Line 4 casts the product of the loop to a list of `tape`
-(string as a character list).
+Line 4 defines the product of the loop as a list of `tape`s. 
```

Comment:
Not sure if I feel comfortable with "defines" here. I'd probably go with the more verbose "define the type of the product of the loop as".

"casts" definitely captures this better, but I guess it's not safe to assume the reader knows what casting it. (Might still be worth mentioning that this is a "cast" though.)

Reply:
They know it's a `:cast` from the keyword in the code, they just don't know what that means.  I don't want to repeat the word as I'm explaining that line.

Your verbose version works too but I think it may be unnecessary--- "a list of tapes" underdefines the product, so the reader can infer it's a product type from the context.

Then again, maybe I am assuming a bit too much about what is readily inferred.

Reply:
Keywords will be removed soon, so then "cast" would never show up.

I guess you're right that it under-defines the product. The product that hasn't even been defined itself.

If we end up getting questions about this in urbit-meta, we can always look at it again.

Reply:
That's a good point about keyword removal.  I'd forgotten about that.

Will it really be called casting after the change, or will only us early adopters call it that?  Will future Martians say "ket-hepping" instead?

Reply:
It'll still be a same thing, which is practically just a cast as we know it from other languages, it just won't be labeled in the compiler as such. Who knows what future martians might call it. If enough old-timers call it what it is, then surely a "cast" will always be understood to be `^-` (or even `^+`).